File: python_code/hello.py
from datetime import datetime
import sys
import caldav

#https://nextcloud.r-podcastdev.link/index.php/apps/calendar/p/zNyH2RE3pLHctoEf

#       "https://nextcloud03.webo.cloud/remote.php/dav/calendars/me@myweb.de/personal/"
#       "https://nextcloud.r-podcastdev.link/remote.php/dav/calendars/rpodcast/wimpys-world-of-streamers/"
#caldav_url = 'https://calendar.example.com/dav'
caldav_url = 'https://nextcloud.r-podcastdev.link/remote.php/dav'

# substitute for real user name and password (preferebly from env vars)
username = 'rpodcast'
password = 'mypassword'

client = caldav.DAVClient(url=caldav_url, username=username, password=password)
my_principal = client.principal()
calendars = my_principal.calendars()

wimpy_cal = "https://nextcloud.r-podcastdev.link/remote.php/dav/calendars/rpodcast/wimpys-world-of-streamers/"

the_same_calendar = client.calendar(url=wimpy_cal)

all_events = the_same_calendar.events()

if all_events:
    print("your calendar has %i cevents:" % len(all_events))
    for e in all_events:
        print(e)
        print(e.data)
else:
    print("your calendar has no events")


if calendars:
    ## Some calendar servers will include all calendars you have
    ## access to in this list, and not only the calendars owned by
    ## this principal.
    print("your principal has %i calendars:" % len(calendars))
    for c in calendars:
        print("    Name: %-20s  URL: %s" % (c.name, c.url))
else:
    print("your principal has no calendars")

File: dev/r_code/hello.R
library(purrr)

# these should not be needed now that I set the python path env var via  .Rprofile
#reticulate::use_python("/usr/bin/python3")
#reticulate::py_config()
#reticulate::py_available()

library(reticulate)

caldav_url = 'https://nextcloud.r-podcastdev.link/remote.php/dav'

caldav <- import("caldav")

client <- caldav$DAVClient(url = caldav_url, username = Sys.getenv("NEXTCLOUD_USER"), password = Sys.getenv("NEXTCLOUD_PASSWORD"))

my_principal <- client$principal()
calendars <- my_principal$calendars()

purrr::walk(calendars, ~{
  message(glue::glue("Name {.x$name} URL {.x$url}"))
})

per_cal_url <- "https://nextcloud.r-podcastdev.link/remote.php/dav/calendars/rpodcast/personal/"
wimpy_cal_url <- "https://nextcloud.r-podcastdev.link/remote.php/dav/calendars/rpodcast/wimpys-world-of-streamers/"
test_cal_url <- "https://nextcloud.r-podcastdev.link/remote.php/dav/calendars/rpodcast/erics-play-calendar/"

wimpy_cal <- client$calendar(url = wimpy_cal_url)
wimpy_events <- wimpy_cal$events()

one_event <- wimpy_events[[1]]


data_obj <- one_event$data

one_event$get_property(prop = "DTSTART")


library(calendar)
ical_example = readLines(system.file("extdata", "example.ics", package = "calendar"))

x <- ic_attributes_vec(x = ical_example)

ics_df <- ic_read(system.file("extdata", "example.ics", package = "calendar"))
ics_df <- ic_read("r_code/wimpys-world-of-streamers-2021-06-08.ics")

library(caldav)

caldav_url = "https://nextcloud.r-podcastdev.link/remote.php/dav/calendars/rpodcast/wimpys-world-of-streamers/"
cal_data <- 
  caldav_get_all_simple_auth(
    url = caldav_url,
    user = Sys.getenv("NEXTCLOUD_USER"),
    password =  Sys.getenv("NEXTCLOUD_PASSWORD")
  )

x <- cal_data$calendar

cat(x, file = "r_code/tmp.ics")


library(purrr)
tidy_cal <- purrr::map(x, ~stringr::str_split(.x, "\\n"))

ic_read("r_code/tmp.ics")


tidy_cal[[1]]
